httpbakery/oven.go

```diff
@@ -0,0 +1,84 @@
+package httpbakery
+
+import (
+	"net/http"
+	"time"
+
+	"golang.org/x/net/context"
+	errgo "gopkg.in/errgo.v1"
+
+	"gopkg.in/macaroon-bakery.v2-unstable/bakery"
+)
+
+// Oven is like bakery.Oven except it provides a method for
+// translating errors returned by bakery.AuthChecker into
+// errors suitable for passing to WriteError.
+type Oven struct {
+	// Oven holds the bakery Oven used to create
+	// new macaroons to put in discharge-required errors.
+	*bakery.Oven
+
+	// AuthnExpiry holds the expiry time of macaroons that
+	// are created for authentication. As these are generally
+	// applicable to all endpoints in an API, this is usually
+	// longer than AuthzExpiry. If this is zero, DefaultAuthnExpiry
+	// will be used.
+	AuthnExpiry time.Duration
```

Comment:
I'm wondering if these should be Spelled AuthorizationExpiry and AuthenticationExpiry.

Reply:
Maybe... I'm not sure I want to spell those names out everywhere though.